feature/experimental/bgp/otg_tests/bgp_2byte_4byte_asn_policy_test/bgp_2byte_4byte_asn_policy_test.go

```diff
@@ -0,0 +1,561 @@
+
+	if deviations.NetworkInstanceTableDeletionRequired(dut) {
+		tablePath := gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(dut)).TableAny()
+		for _, table := range gnmi.LookupAll(t, dut, tablePath.Config()) {
+			if val, ok := table.Val(); ok {
+				if val.GetProtocol() == oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP {
+					gnmi.BatchDelete(resetBatch, gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(dut)).Table(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, val.GetAddressFamily()).Config())
+				}
+			}
+		}
+	}
+	resetBatch.Set(t, dut)
+}
+
+func verifyPeer(t *testing.T, nbr *bgpNbr, dut *ondatra.DUTDevice) {
+	statePath := gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(dut)).Protocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "BGP").Bgp()
+	nbrPath := statePath.Neighbor(nbr.peerIP)
+	glblPath := statePath.Global()
+
+	// Check BGP peerAS from telemetry.
+	peerAS := gnmi.Get(t, dut, nbrPath.State()).GetPeerAs()
+	if peerAS != nbr.peerAS {
+		t.Errorf("BGP peerAs: got %v, want %v", peerAS, nbr.peerAS)
+	}
+
+	// Check BGP localAS from telemetry.
+	localAS := gnmi.Get(t, dut, nbrPath.State()).GetLocalAs()
+	if localAS != nbr.localAS {
+		t.Errorf("BGP localAS: got %v, want %v", localAS, nbr.localAS)
+	}
+
+	// Check BGP globalAS from telemetry.
+	globalAS := gnmi.Get(t, dut, glblPath.State()).GetAs()
+	if globalAS != globalAsNumber {
+		t.Errorf("BGP globalAS: got %v, want %v", globalAS, nbr.localAS)
+	}
+
+	verifyPrefixesTelemetry(t, dut, 3, nbr.isV4)
+}
+
+func configureATE(t *testing.T, ateParams *bgpNbr, connectionType string, prefixes []string, asWidth int) gosnappi.Config {
+	t.Helper()
+	t.Log("Create otg configuration")
+	ate := ondatra.ATE(t, "ate")
+	port1 := ate.Port(t, "port1")
+	config := gosnappi.NewConfig()
+
+	config.Ports().Add().SetName(port1.ID())
+	srcDev := config.Devices().Add().SetName(ateSrc.Name)
+	srcEth := srcDev.Ethernets().Add().SetName(ateSrc.Name + ".Eth").SetMac(ateSrc.MAC)
+	srcEth.Connection().SetPortName(port1.ID())
+	srcIpv4 := srcEth.Ipv4Addresses().Add().SetName(ateSrc.Name + ".IPv4")
+	srcIpv4.SetAddress(ateSrc.IPv4).SetGateway(dutSrc.IPv4).SetPrefix(uint32(ateSrc.IPv4Len))
+	srcIpv6 := srcEth.Ipv6Addresses().Add().SetName(ateSrc.Name + ".IPv6")
+	srcIpv6.SetAddress(ateSrc.IPv6).SetGateway(dutSrc.IPv6).SetPrefix(uint32(ateSrc.IPv6Len))
+
+	srcBgp := srcDev.Bgp().SetRouterId(srcIpv4.Address())
+	if ateParams.isV4 {
+		srcBgpPeer := srcBgp.Ipv4Interfaces().Add().SetIpv4Name(srcIpv4.Name()).Peers().Add().SetName(ateSrc.Name + ".BGP4.peer")
+		srcBgpPeer.SetPeerAddress(ateParams.peerIP).SetAsNumber(ateParams.localAS)
+		if connectionType == connInternal {
+			srcBgpPeer.SetAsType(gosnappi.BgpV4PeerAsType.IBGP)
+		} else {
+			srcBgpPeer.SetAsType(gosnappi.BgpV4PeerAsType.EBGP)
+		}
+		if asWidth == 2 {
+			srcBgpPeer.SetAsNumberWidth(gosnappi.BgpV4PeerAsNumberWidth.TWO)
+		}
+		prefixArr1 := strings.Split(prefixV4[0], "/")
```

Comment:
I think it's worth adding a function here (maybe titled `prefixAndLen`) that returns both the IP string and the length already casted as a uint32.

One nit: I think 'length' is more accurate than 'mask' here.

Reply:
prefixAndLen added